posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -374,7 +408,51 @@ async def materialize_model(model_label: str, team: Team) -> tuple[str, DeltaTab
         prepare_s3_files_for_querying(saved_query.folder_path, saved_query.name, file_uris)
 
     key, delta_table = tables.popitem()
-    return (key, delta_table)
+
+    # Count rows and update both DataWarehouseTable and DataModelingJob
+    row_count = await asyncio.to_thread(count_delta_table_rows, delta_table)
+    await update_table_row_count(saved_query, row_count)
+
+    # Update the job record with the row count and completed status
+    job.rows_materialized = row_count
+    job.status = DataModelingJob.Status.COMPLETED
+    job.last_run_at = dt.datetime.now(dt.UTC)
+    await database_sync_to_async(job.save)()
+
+    return (key, delta_table, job.id)
+
+
+def count_delta_table_rows(delta_table: DeltaTable) -> int:
+    """
+    Count the number of rows in a Delta table from its latest operation.
+
+    The history(1) method returns the most recent operation in the Delta table's
+    transaction log. We use this to get the number of rows added in the latest
+    operation rather than scanning the entire table, which is more efficient.
+    """
+    try:
+        latest_operation = delta_table.history(1)[0]
+        row_count = int(latest_operation["operationMetrics"].get("num_added_rows", 0))
+        return row_count
+    except Exception as e:
+        logger.exception("Failed to count rows in Delta table: %s", str(e))
+        return 0
```

Comment:
Have we seen any errors here? What are we protecting against with the `try/except`? I am not a huge fan of swallowing the error. How do we differentiate between jobs that actually ran 0 rows and jobs that failed to get the count here?

Reply:
I have not, but was planning on monitoring the exception message because was trying to be a bit defensive here, I can remove it if you think it's not feasible.